components/suggest/src/store.rs

```diff
@@ -221,38 +222,48 @@ where
                     .get_records_with_options(&options)?
                     .records;
 
-                self.ingest_records(writer, &records_chunk)?;
+                self.ingest_records(writer, LAST_INGEST_META_UNPARSABLE, &records_chunk)?;
             }
         }
 
+        let ingest_record_types = if let Some(rt) = constraints.allowed_record_types {rt} else {DEFAULT_RECORDS_TYPES.to_vec()};
+        for ingest_record_type in ingest_record_types {
+            self.ingest_records_by_type(ingest_record_type, writer, constraints.max_suggestions)?;
+        }
+
+        Ok(())
+    }
+
+    fn ingest_records_by_type(&self, ingest_record_type: SuggestRecord, writer: &SuggestDb, max_suggestions:Option<u64> ) -> Result<()> {
```

Comment:
One other suggestion for making this a bit more Rusty...

* Let's give `SuggestionProvider` two methods: `get_rs_types()`, which you already added to `SuggestRecord`, and `last_ingest_meta_key()`, to build the meta key string for us.
* `ingest_records_by_type()` can take a `SuggestionProvider`, and use those methods to figure out which records it needs to fetch for that provider, and which key to use for the last ingest time.

Reply:
> , and last_ingest_meta_key(), to build the meta key string for us. 

Good idea!